admin_apps/shared_utils.py

```diff
@@ -829,261 +878,96 @@ def download_yaml(file_name: str, conn: SnowflakeConnection) -> str:
             return yaml_str
 
 
-def unpack_yaml(
-    data: Any | dict[str, Any] | list[Any]
-) -> Any | dict[str, Any] | list[str]:
+def get_sit_query_tag(
+    vendor: Optional[str] = None, action: Optional[str] = None
+) -> str:
     """
-    Recursively unpacks a YAML structure into a Python dictionary.
+    Returns SIT query tag.
+    Returns: str
     """
-    if isinstance(data, dict):
-        return {key: unpack_yaml(value) for key, value in data.items()}
-    elif isinstance(data, list):
-        return [unpack_yaml(item) for item in data]
-    else:
-        return data
 
-
-def load_yaml_file(file_paths: list[Any]) -> list[Any]:  # type: ignore
-    """
-    Loads one or more YAML files and combines them into a single list.
-    """
-    combined_yaml = []
-    for file_path in file_paths:
-        yaml_content = yaml.safe_load(file_path)
-        combined_yaml.append(unpack_yaml(yaml_content))
-    return combined_yaml
+    query_tag = {
+        "origin": "sf_sit",
+        "name": "skimantics",
+        "version": {"major": 1, "minor": 0},
+        "attributes": {"vendor": vendor, "action": action},
+    }
+    return json.dumps(query_tag)
 
 
-def extract_key_values(data: list[dict[str, Any]], key: str) -> list[Any]:
-    """
-    Extracts key's value from a list of dictionaries.
-    """
-    result = []
-    for item in data:
-        values = item.get(key, [])
-        if isinstance(values, list):
-            result.extend(values)
-        else:
-            result.append(values)
-    return result
-
-
-def extract_expressions_from_sections(
-    data_dict: dict[str, Any], section_names: list[str]
-) -> dict[str, dict[str, Any]]:
+def set_sit_query_tag(
+    conn: SnowflakeConnection,
+    vendor: Optional[str] = None,
+    action: Optional[str] = None,
+) -> None:
     """
-    Extracts data in section_names from a dictionary into a nested dictionary:
+    Sets query tag on a single zero-compute ping for tracking.
+    Returns: None
     """
 
-    def extract_key(obj: dict[str, Any]) -> str | Any:
-        return obj.get("expr", obj["name"]).lower()
-
-    d = {}
-    for i in section_names:
-        d[i] = {extract_key(obj): obj for obj in data_dict.get(i, [])}
+    query_tag = get_sit_query_tag(vendor, action)
 
-    return d
+    conn.cursor().execute(f"alter session set query_tag='{query_tag}'")
+    conn.cursor().execute("SELECT 'SKIMANTICS';")
```

Comment:
Is this meant to be hardcoded?

Reply:
Yes, it is. I'm open to alternative implementations of course. Functions set_sit_query_tag and get_sit_query_tag are utility functions explicitly to add Solution Innovation Tag (SIT) query tags that the team uses to track solution usage and consumption. 

`SELECT SKIMANTICS` is just a zero-consumption query on which we will set the query tag that we will use to track customers proceeding through the partner support funnel.

Reply:
Gotcha, I wasn't familiar with this and the query tag, but that context makes sense. Just wanted to double check it wasn't being left in for testing purposes.